src/bank.rs

```diff
@@ -564,11 +591,15 @@ impl Bank {
     }
 
     pub fn get_balance(&self, pubkey: &Pubkey) -> i64 {
+        self.get_account(pubkey).map(|a| a.tokens).unwrap_or(0)
+    }
+
+    pub fn get_account(&self, pubkey: &Pubkey) -> Option<Account> {
         let bals = self
-            .balances
+            .accounts
             .read()
-            .expect("'balances' read lock in get_balance");
-        bals.get(pubkey).cloned().unwrap_or(0)
+            .expect("'accounts' read lock in get_balance");
+        bals.get(pubkey).cloned()
```

Comment:
Get rid of that `cloned()` and return a reference?

Reply:
Its a reference to something we get out of the rwlock.  i dont think it will work.

```
^^^^ borrowed value does not live long enough
```